Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(sf::Vector2i target)
+{	  
+    if(Handle_Ammo())
+    {
+        id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner());
```

Comment:
Mouse clicks are processing by representation module (`Sources\representation\`). When the user clicks the representation calculates which item he clicked (or what interface item). After that, it sends message about click to server. This message will be delivered to the current active mob (for example, to some human). So there are no info for the game where exactly user has clicked, but the game knows which item has been clicked by user.
Functions `posx` and `posy` return the absolute position if the tile. Moreover, we can call functions `GetX` and `GetY` from an object in order to get his absolute position (~= absolute position of his owner - tile). It is better to use `GetX` and `GetY` functions, `posx` and `posy` are kind of deprecated and most likely will be removed.